src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -947,7 +889,7 @@ static term nif_gpio_set_pin_pull(Context *ctx, int argc, term argv[])
     UNUSED(argc);
     UNUSED(argv);
     AVM_LOGW(TAG, "Pull mode must be set using `gpio:set_pin_mode/2` arg #2 i.e. {Mode,PullMode}");
-    return ERROR_ATOM;
+    RAISE_ERROR(UNDEFINED_ATOM);
```

Comment:
Would `FUNCTION_CLAUSE_ATOM` be a better choice here?  I think we are trying to cover the case where a user is calling a function that is not defined for this set of inputs, right?  @pguyot might have some options here, as I think he has suggested the same.

Reply:
I probably made the wrong choice, but I wanted the behavior to be the same as if the function was not defined at all, but still return the warning in the logs explaining the problem, since this driver acts a little differently than the "standard" behavior of the ESP32 and Pico implementations of the driver.

Reply:
I think if the function is undefined you may want `UNDEF_ATOM`